components/hab/src/cli.rs

```diff
@@ -1316,6 +1323,36 @@ fn non_empty(val: String) -> result::Result<(), String> {
     }
 }
 
+/// Adds extra configuration options for shutting down a service with
+/// a customized shutdown signal and timeout.
+///
+/// These are currently feature-flagged. Eventually, we hope to have
+/// this be definable in a package or at load time.
+fn maybe_add_configurable_shutdown_options(mut app: App<'static, 'static>,
+                                           feature_flags: FeatureFlag)
+                                           -> App<'static, 'static> {
+    if feature_flags.contains(FeatureFlag::CONFIGURE_SHUTDOWN) {
+        app = app.arg(Arg::with_name("SHUTDOWN_SIGNAL").help("The signal to send to a service \
```

Comment:
I think we only want this to be a viable option on linux. So only add to clapp on linux platforms.

Reply:
I had that initially, but @baumanj brought up a good point in https://github.com/habitat-sh/habitat/pull/6450#discussion_r276860946; namely, shouldn't Windows `hab` users be able to interact with Linux Supervisors?

I added https://github.com/habitat-sh/habitat/pull/6450/commits/e13c48ad2760de9fb503429b35aff9ac9cdb366d to address this.

Reply:
oh ok that makes sense, but maybe we should add something to the arg help text like "applies to linux supervisors only."

Reply:
oh duh! it does.